include/hjson/hjson.h

```diff
@@ -142,6 +146,14 @@ class Value {
   // Throws if used on VECTOR or MAP
   double to_double() const;
   std::string to_string() const;
+
+  // commentary access functions
+  std::string& comment_pre();
+  std::string& comment_post();
+  std::string& comment_inside();
+  const std::string& comment_pre() const;
+  const std::string& comment_post() const;
+  const std::string& comment_inside() const;
```

Comment:
Should we really return a reference to the internal object? Perhaps someone (by mistake or otherwise) stores the reference, for example `const std::string& mystring = thevalue.comment_inside()`. And then the reference becomes invalid if a new value is assigned to `thevalue`.

Reply:
I use this in the encoder too, so it is a decision between performance and safety again ;-). One should never keep a reference without knowing its lifetime, but of course you are right. I think this decision should depend on the setter decision...

Reply:
I choose safety. :)  Using references internally in the encoder is not a problem, but I prefer to not return references from the public interface.